typing/printtyp.ml

```diff
@@ -624,9 +673,8 @@ let same_printing_env env =
   Env.same_types !printing_old env && Concr.equal !printing_pers used_pers
 
 let set_printing_env env =
-  printing_env := env;
```

Comment:
The root issue is here: both `-short-paths` and the name collision mechanism relied on the ability to update the environment when printing signatures. I think this function should be renamed to `recompute_map`, and the use of `set_printing_env` in `hide_rec_items`  and `still_in_type_group` should be replaced by call to `recompute_map+wrap_env`.

Fixing these two points (see the quick pach [here](https://github.com/Octachron/ocaml/commit/7b744404141502a122546ce197373e081c8950dd)) let me with only the two failing tests in `tool-toplevel` and `tool-debugger` due to a missing_call to `wrap_printing_env`.

Reply:
I wasn't sure of the performance implications of adding more wrap_env, so I just made the behavior closer to the old code